core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4445,6 +4445,7 @@ else if (element instanceof GroupingSets groupingSets) {
                             }
                             case ROLLUP -> groupingSets.getSets().size() + 1;
                             case EXPLICIT -> groupingSets.getSets().size();
+                            case AUTO -> 0;
```

Comment:
This should be 1. AUTO implies a single grouping set made up of all the columns that don't have aggregate operations.